src/Data/CurrentUser/verifySameWhoAmIUser.ts

```diff
@@ -0,0 +1,25 @@
+import { pisaUrl } from '../pisaClient'
+import { WhoAmI } from './CurrentUserDataItem'
+import { simpleErrorToast } from './errorToast'
+
+// TODO: Switch function to pisaClient, once #11616 is resolved
+export async function verifySameWhoAmIUser(
+  email: string,
+  Authorization: string,
+) {
+  const baseUrl = await pisaUrl()
+  if (!baseUrl) return
+
+  const response = await fetch(`${baseUrl}/whoami`, {
+    headers: { Authorization },
+  })
+  if (response.status === 401) simpleErrorToast('URL token is unauthorized')
```

Comment:
* [ ] FYI Do we know more about the reason in order to make this more useful to the visitor? Something like "The link you followed is invalid or has expired."?